src/tools/cephfs/TableTool.cc

```diff
@@ -351,11 +351,18 @@ class TableHandlerOmap
     table_inst.encode_header(&header_bl);
 
     // Compose a transaction to clear and write header
-    librados::ObjectWriteOperation op;
-    op.omap_clear();
-    op.omap_set_header(header_bl);
-    
-    return io->operate(object_name, &op);
+    librados::ObjectWriteOperation clear_op;
+    clear_op.omap_clear();
+    int r = io->operate(object_name, &clear_op); 
+    if (r == -ENOENT) {
+      // Pass: if object doesn't exist then no need to omap_clear it
+    } else if (r < 0) {
+      return r;
+    }
```

Comment:
I'm not sure exactly how to plumb it through the interface, but I think using the FAILOK flag on the omap_clear should be fine and that's what we want to do rather than wait for multiple ops.